### DISCUSSION TITLE
如何在开源项目中做重构？

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#338] [Announcements]
# 如何在开源项目中做重构？

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/2022/01-refactor-in-open-source-project/

**Comment by hualet** [comment]
高效的人总有让自己保证高效的方法，赞～

**Comment by liukun4515** [comment]
这个可以作为：如何参与、深入参与一个开源项目的指导了。
很赞同：在开始的时候，在小的模块和改动刷脸熟；在有大的改动的地方提proposal，以及如何拆分pr、任务的想法。

如何在一个大的proposal中保持专注真的重要了，自己在做的时候发现最有干劲、充实的就是一个大proposal的前两周；后续会因为种种小问题扩展边界，导致“浪费”了一部分专注时间，问题收口也变的困难了。
